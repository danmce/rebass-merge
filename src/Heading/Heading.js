import React from 'react'
import PropTypes from 'prop-types'
import { Heading as RebassHeading } from 'rebass'

const Heading = ({ children, ...props }) => (
  <RebassHeading {...props}>{children}</RebassHeading>
)

Heading.propTypes = {
  children: PropTypes.string,
  fontWeight: PropTypes.oneOf(['bold', 'normal']),
  color: PropTypes.string
}

export default Heading
