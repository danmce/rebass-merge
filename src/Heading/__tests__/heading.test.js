import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Heading from '../Heading'

test('renders Heading correctly', () => {
  const heading = (
    <Heading fontSize={[5, 6, 7]} color='primary'>
      Hello
    </Heading>
  )

  expect(toJson(enzyme.mount(heading))).toMatchSnapshot('enzyme.mount')
})
