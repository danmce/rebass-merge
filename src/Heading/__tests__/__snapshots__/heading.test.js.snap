// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Heading correctly: enzyme.mount 1`] = `
<Heading
  color="primary"
  fontSize={
    Array [
      5,
      6,
      7,
    ]
  }
>
  <ForwardRef
    color="primary"
    fontSize={
      Array [
        5,
        6,
        7,
      ]
    }
  >
    <Styled(div)
      __css={
        Object {
          "fontFamily": "heading",
          "fontSize": 4,
          "fontWeight": "heading",
          "lineHeight": "heading",
        }
      }
      as="h2"
      color="primary"
      fontSize={
        Array [
          5,
          6,
          7,
        ]
      }
      tx="text"
      variant="heading"
    >
      <h2
        className="emotion-0"
      >
        Hello
      </h2>
    </Styled(div)>
  </ForwardRef>
</Heading>
`;
